app/test_main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from unittest import mock
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_can_access_google_page(
+        mock_has_internet_connection: bool,
+        mock_valid_google_url: bool
+) -> None:
+    mock_valid_google_url.return_value = True
+    mock_has_internet_connection.return_value = True
+
+    assert can_access_google_page("https://www.google.com") == "Accessible"
+
+    mock_valid_google_url.return_value = False
+    assert can_access_google_page("https://invald-url.com") == "Not accessible"
+
+    mock_valid_google_url.return_value = True
+    mock_has_internet_connection.return_value = False
+    assert can_access_google_page("https://www.google.com") == "Not accessible"
```

Comment:
By using @pytest.mark.parametrize, you can create more concise and efficient test cases that test various scenarios without duplication